chain/client/src/sync/state.rs

```diff
@@ -1266,6 +1276,33 @@ fn request_part_from_peers(
     );
 }
 
+/// Takes a received state part and attempts to validate and store the part.
+/// Used both for parts downloaded from external storage and parts received from peers.
+/// This process is slow and should only occur on state_parts_future_spawner.
+/// Returns a result to be passed back via state_parts_mpsc_tx.
+fn try_validate_and_store_received_state_part(
```

Comment:
not important to do this here, but it seems this is the exact same thing as `Chain::set_state_part()` but returning the data length, and just a string for an error? This should prob just deleted eventually, imo. But since this is all getting refactored anyway it's not a big deal

Reply:
Yeah I noticed this as well and ideally we would not duplicate this logic between here and `Chain:: set_state_part`. The part I had trouble with is there's no issue cloning and passing an `Arc<dyn RuntimeAdapter>` to these futures but we cannot pass all of chain.